doc/tutorials/troubleshooting.rst

```diff
@@ -0,0 +1,47 @@
+Troubleshooting
+===============
+
+Plugin errors
+-------------
+
+Errors that start with ``GHC Core to PLC plugin`` are ``plutus-tx-plugin`` errors.
+The most common error is:
+
+``Error: Reference to a name which is not a local, a builtin, or an external INLINABLE function``
+
+This means the plugin doesn't have access to implementation of the function, which it needs to be able to compile the function to Plutus Core.
+Some things you can do to fix it:
+
+- Make sure to add ``{-# INLINABLE functionname #-}`` to your function.
+- If there's an extra ``$c`` in front of the function name in the error, GHC has generated a specialised version of your function,
+  which prevents the plugin from accessing it.
+  You can turn it off with ``{-# OPTIONS_GHC -fno-specialise #-}``
+- Other compiler options that could fix the issue are
+
+  - ``{-# OPTIONS_GHC -fno-strictness #-}``
+  - ``{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}``
+  - ``{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}``
+  - ``{-# OPTIONS_GHC -fobject-code #-}``
+
+  Some more details are in `the plutus-tx readme <https://github.com/input-output-hk/plutus/tree/master/plutus-tx#building-projects-with-plutus-tx>`_.
+
+If you don't need the plugin to succeed, f.e. when Haddock is building documentation,
+you can pass the flag ``-fdefer-plugin-errors``.
+
+.. note::
+  The recommended way to build documentation is with ``nix-build default.nix -A docs.site``
```

Comment:
Does this belong here?

Reply:
My thinking was that if you encounter this issue, it might very well be because you are trying to build haddock documentation. In normal usage this flag should not be needed?

Reply:
This is documentation for our users: I doubt that they're going to be doing that. Such advice might belong in `CONTRIBUTING`, which is for us!

Reply:
Ah right. This applies to the whole `-fdefer-plugin-errors` part then?

Reply:
Probably. Unless people are doing it for debugging (that's why I added it!) or reusing our Haddock workaround. But hopefully the Haddock issues will be improved by then..